### PR TITLE
tests(events): migrate flaky event simulation tests to new format

### PR DESCRIPTION
**Pull request by glevco**
Some event tests were skipped in https://github.com/HathorNetwork/hathor-core/pull/646 for not working correctly. This PR reimplements those tests using the new format introduced by https://github.com/HathorNetwork/hathor-core/pull/647.

### Acceptance Criteria

- Remove previous event scenarios simulation tests and create new ones
- Remove `DummyMiner`, as it was probably the cause for the faulty tests, and using simulator triggers is a much better approach

### CODE REVIEW

File: hathor/simulator/miner/dummy_miner.py
Comment (glevco) [inline hathor/simulator/miner/dummy_miner.py:1]: @msbrogli I remember that when I created the `DummyMiner`, I implemented the `AbstractMiner` base class for it, abstracting methods from the existing miner simulator, and you raised questions about its necessity. Now that `DummyMiner` is removed, there's only one miner simulator that inherits from `AbstractMiner`, so we could remove it too. What do you think?

Comment (msbrogli) [reply hathor/simulator/miner/dummy_miner.py:1]: I'm ok either removing or keeping it. I'm more curious on why the `DummyMiner` was causing random test failures.

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — APPROVED** [review]

File: hathor/simulator/miner/dummy_miner.py
Comment (glevco) [reply hathor/simulator/miner/dummy_miner.py:1]: I'm also curious, but unfortunately I can't pinpoint the exact reason. It was some weird behavior related to how the code interacted with twisted, that I can only assume was originated on the DummyMiner. When I noticed that using triggers was a much better option, I tested and it worked, so I didn't invest time investigating the DummyMiner. There's nothing it does that can't be done with triggers, so there's no reason to keep it as it has been proven faulty, and I don't see a lot of benefit on investigating it further

Comment (msbrogli) [reply hathor/simulator/miner/dummy_miner.py:1]: I agree it's not worth diving deep. Let's just remove it to prevent others from using it.

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — APPROVED** [review]

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — APPROVED** [review]

**Pull request merged**